TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -2,169 +2,132 @@ const Parser = require( './Parser' );
-            let curMetricValues = null;
-            let curProductResource = null;
-            let curRegex = null;
-            let isValid = true;
-            let curTestData = {};
-
-            // Parse benchmark name
-            if ( ( curRegexResult = benchmarkNameRegex.exec( curItr.value ) ) !== null ) {
-                curBenchmarkName = curRegexResult[1];
+        if ((tests.map(x=>x.testResult).indexOf("PASSED") > -1)) {
+            if ((tests.map(x=>x.testResult).indexOf("FAILED") > -1)) {
+                buildResult = "PARTIAL-SUCCESS";
             } else {
-                isValid = false;
+                buildResult = "SUCCESS";
             }
+        } else {
+            buildResult = "FAILURE";
+        }
+        return buildResult;
+    }
 
-            // Parse benchmark variant
-            if ( ( curRegexResult = benchmarkVariantRegex.exec( curItr.value ) ) !== null ) {
-                curBenchmarkVariant = curRegexResult[1];
-            } else {
-                isValid = false;
+    static parsePerf(results) {
+        if(Array.isArray(results) && results.length > 0) {
+            for(let result of results) {
+                if (this.canParse(result.testName)) {
+                    const {benchmarkName, benchmarkVariant, testData} = this.parse(result.testName, result.testOutput, result.testResult);
+                    result.benchmarkName = benchmarkName;
+                    result.benchmarkVariant = benchmarkVariant;
+                    result.testData = testData;
+                }
             }
+        }
+        return results;
+    }
 
-            if (!BenchmarkMetricRouter[curBenchmarkName]) {
-                isValid = false;
-            } else if (!BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]) {
-                isValid = false;
-            // Benchmark should have at least one metric to parse with
-            } else if (!BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]
-            || !BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]
-            || Object.keys(BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]).length === 0) {
-                isValid = false;
-            }
-    
-            if ( ( curRegexResult = productResourceRegex.exec( curItr.value ) ) !== null ) {
-                curProductResource = curRegexResult[1];
-            }
+    static parse( testName, testOutput ) {
+        const {benchmarkName, benchmarkVariant} = this.getBenchmarkInfo(testName);
+        let isValid = true;
+        let curTestData = {};
+        let curBenchVariant = null;
+        let curMetric = null;
+        let curRegexResult = null;
+        let curMetricValues = null;
+        let curRegex = null;
+
+        if ( !BenchmarkMetric[BenchmarkMetricRouter[benchmarkName][benchmarkVariant]] || !BenchmarkMetric[BenchmarkMetricRouter[benchmarkName]["0"]]
```

Comment:
Doesn't look right! A test can't have both `benchmarkVariant` and `0`, so can't put them in one check. 
Won't this make a TKG build invalid always? Same thing for `metrics` check as well. We'll need something similar to `const variantValue = parseInt(benchmarkVariant);` and then use `isNaN` as done in `canParse()`.